cmd/truss/main.go

```diff
@@ -24,22 +24,45 @@ import (
 )
 
 var (
-	pbPackageFlag  = flag.String("pbout", "", "The go package path where the protoc-gen-go .pb.go structs will be written.")
-	svcPackageFlag = flag.String("svcout", "", "The go package path where the generated service directory will be written.")
-	verboseFlag    = flag.Bool("v", false, "Verbose output with stack traces for errors.")
+	pbPackageFlag  = flag.String("pbout", "", "Go package path where the protoc-gen-go .pb.go files will be written")
+	svcPackageFlag = flag.String("svcout", "", "Go package path where the generated Go service will be written")
+	verboseFlag    = flag.Bool("v", false, "verbose output")
+)
+
+var binName = filepath.Base(os.Args[0])
+
+const (
+	noVersion   string = "<no-version>"
+	noBuildDate string = "<no-build-date>"
+)
+
+var (
+	Version   string = noVersion
+	BuildDate string = noBuildDate
 )
 
 func init() {
+	var buildinfo string
+	if Version != "" && Version != noVersion {
+		buildinfo = fmt.Sprintf("version: %s", Version)
+	}
+	if BuildDate != "" && BuildDate != noBuildDate {
+		buildinfo = fmt.Sprintf("%s built: %s", buildinfo, strings.Replace(BuildDate, "_", " ", -1))
+	}
+
 	flag.Usage = func() {
-		fmt.Fprintf(os.Stderr, "Usage: %s [OPTION]... [*.proto]...\n", filepath.Base(os.Args[0]))
+		if buildinfo != "" {
+			fmt.Fprintf(os.Stderr, "%s (%s)\n", binName, strings.TrimSpace(buildinfo))
+		}
+		fmt.Fprintf(os.Stderr, "Usage: %s [OPTIONS] protofile [protofile2...protofileN]\n", binName)
```

Comment:
I tried to simplify this a little, but didn't do great. We should make it comply with this http://docopt.org/, which a hybrid of the before and after of this line comes close to.